### PR TITLE
Add Linea to price api supported chains

### PR DESCRIPTION
**Pull request by bergeron**
## Explanation

Adds Linea to the list of chains the price API supports.  To enable pricing ERC20 tokens.

Before:

Screenshot 2024-01-18 at 12 03 53 PM

After:

Screenshot 2024-01-18 at 3 50 05 PM

### `@metamask/assets-controllers`

- **ADDED**: Add Linea to the price api supported chains

## Checklist

- [ ] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/assets-controllers/src/token-prices-service/codefi-v2.ts [hunk 1]
```diff
@@ -233,6 +233,8 @@ export const SUPPORTED_CHAIN_IDS = [
   '0x4e454152',
   // Harmony Mainnet Shard 0
   '0x63564c40',
+  // Linea Mainnet
+  '0xe708'
```
Comment (bergeron) [inline packages/assets-controllers/src/token-prices-service/codefi-v2.ts:237]: It looks to me like many chains in this list aren't actually supported?  Maybe I'm missing something.

https://github.com/MetaMask/core/pull/3600#discussion_r1412568649 suggest it was sourced from some mobile code.  But if you call some of them:

`'321' is not a valid chainId query parameter. Must be a number of the set: [1, 10, 56, 100, 137, 250, 324, 8453, 42161, 43114, 59144]`

Which matches the price API source code's [`supportedChains`](https://github.com/consensys-vertical-apps/va-mmcx-price-api/blob/8dfb193fb7c64dc7ee8c31dc238a20a51b83bc3d/src/server/price/constants.ts#L203)


And a final test:
```
const SUPPORTED_CHAIN_IDS = [
    '0x1','0xa','0x19','0x38','0x39','0x42','0x46','0x52','0x58','0x64',
    '0x6a','0x7a','0x80','0x89','0xfa','0x120','0x141','0x144','0x169',
    '0x440','0x504','0x505','0x2105','0x150','0x2710','0xa4b1','0xa4ec',
    '0xa516','0xa86a','0x518af','0x4e454152','0x63564c40'
].map(c => parseInt(c, 16));

const actuallySupported = [];

for (const chain of SUPPORTED_CHAIN_IDS) {
    const resp = await (await fetch(
        `https://price-api.metafi.codefi.network/v2/chains/${chain}/spot-prices?tokenAddresses=0x0000000000000000000000000000000000000000`
    )).json();

    if (resp?.statusCode !== 400) {
        actuallySupported.push(chain);
    }
}

console.log(actuallySupported)

// [1, 10, 56, 100, 137, 250, 324, 8453, 42161, 43114]
```

Which is much smaller.  And Linea is the only supported one we're missing.

File: packages/assets-controllers/src/token-prices-service/codefi-v2.ts [hunk 2]
```diff
@@ -233,6 +233,8 @@ export const SUPPORTED_CHAIN_IDS = [
   '0x4e454152',
   // Harmony Mainnet Shard 0
   '0x63564c40',
+  // Linea Mainnet
+  '0xe708',
```
Comment (bergeron) [inline packages/assets-controllers/src/token-prices-service/codefi-v2.ts:237]: It looks to me like many chains in this list aren't actually supported?  Maybe I'm missing something.

https://github.com/MetaMask/core/pull/3600#discussion_r1412568649 suggest it was sourced from some mobile code.  But if you call some of them:

`'321' is not a valid chainId query parameter. Must be a number of the set: [1, 10, 56, 100, 137, 250, 324, 8453, 42161, 43114, 59144]`

Which matches the price API source code's [`supportedChains`](https://github.com/consensys-vertical-apps/va-mmcx-price-api/blob/8dfb193fb7c64dc7ee8c31dc238a20a51b83bc3d/src/server/price/constants.ts#L203)


And a final test:
```
const SUPPORTED_CHAIN_IDS = [
    '0x1','0xa','0x19','0x38','0x39','0x42','0x46','0x52','0x58','0x64',
    '0x6a','0x7a','0x80','0x89','0xfa','0x120','0x141','0x144','0x169',
    '0x440','0x504','0x505','0x2105','0x150','0x2710','0xa4b1','0xa4ec',
    '0xa516','0xa86a','0x518af','0x4e454152','0x63564c40'
].map(c => parseInt(c, 16));

const actuallySupported = [];

for (const chain of SUPPORTED_CHAIN_IDS) {
    const resp = await (await fetch(
        `https://price-api.metafi.codefi.network/v2/chains/${chain}/spot-prices?tokenAddresses=0x0000000000000000000000000000000000000000`
    )).json();

    if (resp?.statusCode !== 400) {
        actuallySupported.push(chain);
    }
}

console.log(actuallySupported)

// [1, 10, 56, 100, 137, 250, 324, 8453, 42161, 43114]
```

Which is much smaller.  And Linea is the only supported one we're missing.

Comment (mcmire) [reply packages/assets-controllers/src/token-prices-service/codefi-v2.ts:237]: Ah, okay. I didn't realize there was an official list somewhere, I did indeed copy that list from mobile when I added it, so I'm not sure where they got that from. I'll make another PR which updates this list to match the real thing.

**Review by mcmire — APPROVED** [review]
Looks good!

**Pull request merged**